src/test/unit/profile-tree.test.js

```diff
@@ -225,6 +262,47 @@ describe('unfiltered call tree', function() {
       });
     });
   });
+
+  /**
+   * While not specifically part of the call tree, this is a core function
+   * to help navigate stacks through a list of functions.
+   */
+  describe('getFuncStackFromFuncArray', function() {
+    const profile = getProfileForUnfilteredCallTree();
+    const [thread] = profile.threads;
+    const { funcStackTable } = getFuncStackInfo(
+      thread.stackTable,
+      thread.frameTable,
+      thread.funcTable
+    );
+
+    // Helper to make the assertions a little less verbose.
+    function checkStack(funcArray, index, name) {
+      it(`finds stack that ends in ${name}`, function() {
+        expect(getFuncStackFromFuncArray(funcArray, funcStackTable)).toBe(
+          index
+        );
+      });
+    }
+
+    checkStack([A], A, 'A');
+    checkStack([A, B], B, 'B');
+    checkStack([A, B, C], C, 'C');
+    checkStack([A, B, C, D], D, 'D');
+    checkStack([A, B, C, D, E], E, 'E');
```

Comment:
So getFuncStackInfo here just happens to create exactly one funcStack per func, so all func indices can be used as funcStack indices as well?